### DISCUSSION TITLE
Out of tree sensor driver development

### DISCUSSION THREAD

**Discussion by mithi25** [zephyrproject-rtos/zephyr#54470] [Q&A]
Hi, 

I am trying to develop a out of tree driver for LPS22HB1(this driver is already present in zephyr, so i renamed  it). I have also downloaded and build the example-application successfully from https://github.com/zephyrproject-rtos/example-application.

I added the driver and binding file of LPS22HB1 by taking reference from LPS22HB sensor but getting below errors-
![image](https://user-images.githubusercontent.com/112639800/216902967-a38c79aa-a978-4313-8fa3-a20366ec5db3.png)
![error2](https://user-images.githubusercontent.com/112639800/216903058-8a9fc966-7707-4907-a637-1c24c892e771.PNG)

Kconfig file for the lps22hb1:
menuconfig LPS22HB1
	bool "LPS22HB pressure and temperature"
	default y
	depends on DT_HAS_ST_LPS22HB_PRESS_ENABLED
	select I2C
	help
	  Enable driver for LPS22HB I2C-based pressure and temperature
	  sensor.

config LPS22HB1_SAMPLING_RATE
	int "Output data rate"
	default 25
	depends on LPS22HB1
	help
	  Sensor output data rate expressed in samples per second.
	  Data rates supported by the chip are 1, 10, 25, 50, 75.

Where DT_HAS_ST_LPS22HB_PRESS_ENABLED is declared, I am trying to develop new sensor driver but no idea where to declare this macro. 

Thanks

**Comment by nixward** [comment]
Hi @mithi25,

You'll likely need to either fix the DeviceTree component of your current board definition to add the node that defines the sensor OR add a DeviceTree overlay to add a node that defines the sensor.  See here for how to add a sensor with a overlay: https://docs.zephyrproject.org/3.2.0/build/dts/howtos.html#use-devicetree-overlays see the my_spi_device node label.  You also maybe able to find a similar node definition for a LPS22HB1 sensor in the Zephyr code base.

**Reply by mithi25** [reply]
hi @nixward,

I have added overlay file and it is defines as below:
&i2c0 {
   pinctrl-0 = <&i2c0_default_alt>;
   pinctrl-1 = <&i2c0_sleep_alt>;
   pinctrl-names = "default", "sleep";
    lps22hb1@5C {
            status = "okay";
            compatible = "st,lps22hb1-press";
            reg = <0x5C>;
    };
}
When I am using "select  DT_HAS_ST_LPS22HB_PRESS_ENABLED" instead of "depends on DT_HAS_ST_LPS22HB_PRESS_ENABLED" then code build successfully and working.
menuconfig LPS22HB1
	bool "LPS22HB pressure and temperature"
	default y
	select DT_HAS_ST_LPS22HB_PRESS_ENABLED
	#depends on DT_HAS_ST_LPS22HB_PRESS_ENABLED
	select I2C
	help
	  Enable driver for LPS22HB I2C-based pressure and temperature
	  sensor.

But in Zephyr code base it is using "depends on DT_HAS_ST_LPS22HB_PRESS_ENABLED", So can you please tell me where DT_HAS_ST_LPS22HB_PRESS_ENABLED is defined in zephyr code base so that i can look the definition of it.
